frame/clearing-house/src/lib.rs

```diff
@@ -153,7 +186,21 @@ pub mod pallet {
 	// ----------------------------------------------------------------------------------------------------
 
 	// Helper functions - core functionality
-	impl<T: Config> Pallet<T> {}
+	impl<T: Config> Pallet<T> {
+		#[allow(unused_variables)]
+		pub fn do_add_margin(
+			acc: &T::AccountId,
+			asset: AssetIdOf<T>,
+			amount: T::Balance,
+		) -> Result<(), DispatchError> {
+			if !CollateralTypes::<T>::contains_key(asset) {
+				return Err(Error::<T>::UnsupportedCollateralType.into())
```

Comment:
What do you think sending more informative messages in case of errors?

Something similar to Drift's [`print_error` macro](https://github.com/drift-labs/protocol-v1/blob/master/programs/clearing_house/src/error.rs).

The idea behind `ErrorCode` enum would be good too, I guess.
Thoughs?

Reply:
That sounds good, although I don't know if there's something analogous to that in Substrate. The [docs](https://docs.substrate.io/v3/runtime/events-and-errors/#errors) don't say much. But thanks for the input